paymentsheet/src/main/java/com/stripe/android/customersheet/StripeCustomerAdapter.kt

```diff
@@ -140,12 +140,24 @@ internal class StripeCustomerAdapter @Inject constructor(
                     cachedCustomerEphemeralKey.date
                 )
             }?.result ?: run {
+                val ephemeralKey = customerEphemeralKeyProvider.provideCustomerEphemeralKey()
+
+                ephemeralKey.getOrNull()?.let {
+                    customerRepository.retrieveCustomer(
```

Comment:
This seems unfortunate to have to make a network call to validate this.

I'd imagine this would be an edge case, but we're making everyone pay the penalty of the network call for it.